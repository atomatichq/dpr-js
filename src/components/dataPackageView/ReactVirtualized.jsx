import React, { PropTypes } from 'react'
import { Column, Table } from 'react-virtualized'
import 'react-virtualized/styles.css'
const Spinner = require('react-spinkit')

class ReactVirtualized extends React.Component {

  constructor(props) {
    super(props)
    this.state = {
      data: this.props.spec.data,
      headers: this.props.spec.headers,
      width: this.props.spec.width,
      height: this.props.spec.height,
      headerHeight: this.props.spec.headerHeight,
      rowHeight: this.props.spec.rowHeight,
      rowCount: this.props.spec.rowCount,
      columnWidth: this.props.spec.columnWidth
    }
  }

  componentWillReceiveProps(nextProps) {
    this.setState({
      data: nextProps.spec.data,
      rowCount: nextProps.spec.rowCount,
      height: nextProps.spec.height
    })
  }

  render() {
    let columns = []
    this.state.headers.forEach((header, idx) => {
      columns.push(
        <Column
          label={header}
          dataKey={header}
          width={this.state.columnWidth}
          key={idx}
        />
      )
    })
    return (
      <Table
        className="table"
        headerClassName="rv-header"
        width={this.state.width}
        height={this.state.height}
        headerHeight={this.state.headerHeight}
        rowHeight={this.state.rowHeight}
        rowCount={this.state.rowCount}
        rowGetter={({ index }) => this.state.data[index]}
      >
        {columns}
      </Table>
    )
  }

}

export default ReactVirtualized
