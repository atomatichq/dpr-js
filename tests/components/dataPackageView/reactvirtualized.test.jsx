import 'babel-polyfill'
import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import ReactVirtualized from '../../../src/components/dataPackageView/ReactVirtualized'

const mockSpec = {
  data: [
    {
      "Date": "2014-01-01",
      "High": 14.59,
      "Open": 14.32
    },
    {
      "Date": "2014-01-02",
      "High": 14.22,
      "Open": 14.06
    },
    {
      "Date": "2014-01-05",
      "High": 14,
      "Open": 13.41
    }
  ]
  , headers: [
    'Date'
    , 'Open'
    , 'High'
  ]
  , width: 1136
  , height: 30 * 3 + 20
  , headerHeight: 20
  , rowHeight: 30
  , rowCount: 3
  , "columnWidth": 378.6666666666667
}

const mockSpecNoData = {
  data: undefined
  , headers: [
    'Date'
    , 'Open'
    , 'High'
  ]
  , width: 1136
  , height: 20
  , headerHeight: 20
  , rowHeight: 30
  , rowCount: 0
  , "columnWidth": 378.6666666666667
}

describe('ReactVirtualized component', () => {
  it('should match the snapshot - with data', () => {
    const wrapper = shallow(<ReactVirtualized spec={mockSpec} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should match the snapshot - without data', () => {
    const wrapper = shallow(<ReactVirtualized spec={mockSpecNoData} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
