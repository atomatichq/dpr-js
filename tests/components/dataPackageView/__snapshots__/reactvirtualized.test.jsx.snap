exports[`ReactVirtualized component should match the snapshot - with data 1`] = `
<Table
  className="table"
  disableHeader={false}
  estimatedRowSize={30}
  headerClassName="rv-header"
  headerHeight={20}
  headerRowRenderer={[Function]}
  headerStyle={Object {}}
  height={110}
  noRowsRenderer={[Function]}
  onRowsRendered={[Function]}
  onScroll={[Function]}
  overscanIndicesGetter={[Function]}
  overscanRowCount={10}
  rowCount={3}
  rowGetter={[Function]}
  rowHeight={30}
  rowRenderer={[Function]}
  rowStyle={Object {}}
  scrollToAlignment="auto"
  scrollToIndex={-1}
  style={Object {}}
  width={1136}>
  <Column
    cellDataGetter={[Function]}
    cellRenderer={[Function]}
    dataKey="Date"
    flexGrow={0}
    flexShrink={1}
    headerRenderer={[Function]}
    label="Date"
    style={Object {}}
    width={378.6666666666667} />
  <Column
    cellDataGetter={[Function]}
    cellRenderer={[Function]}
    dataKey="Open"
    flexGrow={0}
    flexShrink={1}
    headerRenderer={[Function]}
    label="Open"
    style={Object {}}
    width={378.6666666666667} />
  <Column
    cellDataGetter={[Function]}
    cellRenderer={[Function]}
    dataKey="High"
    flexGrow={0}
    flexShrink={1}
    headerRenderer={[Function]}
    label="High"
    style={Object {}}
    width={378.6666666666667} />
</Table>
`;

exports[`ReactVirtualized component should match the snapshot - without data 1`] = `
<Table
  className="table"
  disableHeader={false}
  estimatedRowSize={30}
  headerClassName="rv-header"
  headerHeight={20}
  headerRowRenderer={[Function]}
  headerStyle={Object {}}
  height={20}
  noRowsRenderer={[Function]}
  onRowsRendered={[Function]}
  onScroll={[Function]}
  overscanIndicesGetter={[Function]}
  overscanRowCount={10}
  rowCount={0}
  rowGetter={[Function]}
  rowHeight={30}
  rowRenderer={[Function]}
  rowStyle={Object {}}
  scrollToAlignment="auto"
  scrollToIndex={-1}
  style={Object {}}
  width={1136}>
  <Column
    cellDataGetter={[Function]}
    cellRenderer={[Function]}
    dataKey="Date"
    flexGrow={0}
    flexShrink={1}
    headerRenderer={[Function]}
    label="Date"
    style={Object {}}
    width={378.6666666666667} />
  <Column
    cellDataGetter={[Function]}
    cellRenderer={[Function]}
    dataKey="Open"
    flexGrow={0}
    flexShrink={1}
    headerRenderer={[Function]}
    label="Open"
    style={Object {}}
    width={378.6666666666667} />
  <Column
    cellDataGetter={[Function]}
    cellRenderer={[Function]}
    dataKey="High"
    flexGrow={0}
    flexShrink={1}
    headerRenderer={[Function]}
    label="High"
    style={Object {}}
    width={378.6666666666667} />
</Table>
`;
